starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java

```diff
@@ -30,12 +37,37 @@
 @Singleton
 public class HttpClient implements DefaultFeature {
 
+    private final RxJavaTwoHttpClient rxJavaTwoHttpClient;
+    private final RxJavaThreeHttpClient rxJavaThreeHttpClient;
+    private final ReactorHttpClient reactorHttpClient;
+
+    public HttpClient(RxJavaTwoHttpClient rxJavaTwoHttpClient,
+                      RxJavaThreeHttpClient rxJavaThreeHttpClient,
+                      ReactorHttpClient reactorHttpClient) {
+        this.rxJavaTwoHttpClient = rxJavaTwoHttpClient;
+        this.rxJavaThreeHttpClient = rxJavaThreeHttpClient;
+        this.reactorHttpClient = reactorHttpClient;
+    }
+
     @Override
     public boolean shouldApply(ApplicationType applicationType, Options options, Set<Feature> selectedFeatures) {
         return applicationType == ApplicationType.DEFAULT &&
                 selectedFeatures.stream().noneMatch(feature -> feature instanceof FunctionFeature);
     }
 
+    @Override
+    public void processSelectedFeatures(FeatureContext featureContext) {
+        if (featureContext.isPresent(RxJavaTwo.class)) {
+            featureContext.addFeature(rxJavaTwoHttpClient);
+        }
+        if (featureContext.isPresent(RxJavaThree.class)) {
+            featureContext.addFeature(rxJavaThreeHttpClient);
+        }
+        if (featureContext.isPresent(Reactor.class)) {
+            featureContext.addFeature(reactorHttpClient);
+        }
+    }
```

Comment:
I think we should only allow to add one of these features. It doesn't make sense to have rxjava1, 2, 3 and reactor on the same app. Specially for RxJava.

Reply:
But it should be part of a different PR right? If you add RxJava2 and you have other RxJava3 throw an exception.

Reply:
I don't see why. As part of adding a new feature we need to check the other features that are not compatible with the new one.